### PR TITLE
Add HawtioExtension shim

### PR DESCRIPTION
**Pull request by benjaminapetersen**
Fixes [Trello issue](https://trello.com/c/lDwZBWL4/251-free-int-error-loading-web-console-after-upgrade-to-310)

Adds a shim for `HawtioExtension.add()` to map to `extensionRegistry.add()`.  API is fairly similar, however `extensionRegistry` provides `args` to the callback function rather than reference to raw `$scope`.  

The shim will also log a deprecation notice:

screen shot 2018-05-14 at 12 20 50 pm

@spadgett @TheRealJon @jupierce

### CODE REVIEW

**Comment by benjaminapetersen** [comment]
@damemi this will prob give you direction for online update, shouldn't be super significant code changes.

File: app/scripts/services/shims/hawtioExtension.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+     console.warn('HawtioExtension.add() has been deprecated.  Please migrate to angular-extension-registry https://github.com/openshift/angular-extension-registry');
+     return {
+       add: function(endpointName, domNodeGenerator) {
+         extensionRegistry.add(endpointName, function(args) {
+           return {
+             type: 'dom',
+             node: domNodeGenerator(args)
+           }
+         });
+       }
```
Comment (spadgett) [inline app/scripts/services/shims/hawtioExtension.js]: missing `;`

**Comment by TheRealJon** [comment]
lgtm

File: app/scripts/services/shims/hawtioExtension.js (hunk 1 above)
Comment (benjaminapetersen) [reply app/scripts/services/shims/hawtioExtension.js]: @spadgett  I tested with some of that code, yes.  I just swapped to points like `'nav-help-dropdown'`.

Comment (benjaminapetersen) [reply app/scripts/services/shims/hawtioExtension.js]: `;` is our `jshint` job not running?  I don't get any errors locally..

**Comment by spadgett** [comment]
@benjaminapetersen can you update with the semicolon fix so we can merge? :)

**Comment by benjaminapetersen** [comment]
Oop, sorry didn't push :/
Updated

**Comment by spadgett** [comment]
@benjaminapetersen vendor diff :/

**Comment by benjaminapetersen** [comment]
Hmm. 😐

**Comment by benjaminapetersen** [comment]
Ah, good, there we go.

**Pull request merged**